BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Antenna_Performance_Metric/XFintoPUEO.py

```diff
@@ -0,0 +1,345 @@
+        print("Done with vh_0 for Antenna " +str(antenna+1))
+
+# for hv_0:
+for antenna in range(g.NPOP):
+    with open(f'{g.WorkingDir}/Test_Outputs/'+"hv_0_"+{g.gen}+" "+str(antenna+1), "w+") as PUEO_hv0:
+        os.chmod(f'{g.WorkingDir}/Test_Outputs/'+"hv_0_"+{g.gen}+" "+str(antenna+1), 0o777)
+        uanMaxG = []
+        for freq in range(numFreq):
+            uanDatG = readFile(antenna+1, freq+1) #Gives Gain Column           
+            MaxGain = uanDatG[1314] #Gain at theta = 90                        
+            uanMaxG.append(MaxGain)
+
+        OutputHV = np.column_stack((freqArray, uanMaxG))
+        for p in range(numFreq):
+            for q in range(2):
+                PUEO_hv0.write(str(OutputHV[p][q])+" ")
+            PUEO_hv0.write("\n")
+        print("Done with hv_0 for Antenna " +str(antenna+1))
+
+# Now to pull angles 5, 10, 20, 30, 45, 90 for el or az and respective theta or phi
+# Angles and index locations. For theta. 5:[73], 10:[146], 20:[292], 30:[438], 45:[657], 90:[1314]
+# For phi. 5:[1], 10:[2], 20:[4], 30:[6], 45:[9], 90:[18]
+
+# Creating new Array with Frequencies 6 times
+BIGfreqArray = []
+for z in range(6):
+    for q in range(numFreq):
+        scinoteval = "{:.2e}".format(freqVals[q])
+        BIGfreqArray.append(scinoteval)
+numBIGarray = 360
+# for vv_el: (theta)
+
+for antenna in range(g.NPOP):
+    with open(f'{g.WorkingDir}/Test_Outputs/'+"vv_el_"+{g.gen}+" "+str(antenna+1), "w+") as PUEO_vve: 
+        os.chmod(f'{g.WorkingDir}/Test_Outputs/'+"vv_el_"+{g.gen}+" "+str(antenna+1), 0o777)
+        uanAngG = []
+        #print("Gain for 5")
+        for freq in range(numFreq):
+            uanDatG = readFile(antenna+1, freq+1) #Gain Column
+            fiveGain = uanDatG[73]
+            uanAngG.append(fiveGain)
+            #print(fiveGain)
+        #print("Gain for 10")
+        for freq in range(numFreq):
+            uanDatG = readFile(antenna+1, freq+1)
+            tenGain = uanDatG[146]
+            uanAngG.append(tenGain)
+            #print(tenGain)
+        #print("Gain for 20") 
+        for freq in range(numFreq):
+            uanDatG = readFile(antenna+1, freq+1)
+            twentyGain = uanDatG[292]
+            uanAngG.append(twentyGain)
+            #print(twentyGain)
+        #print("Gain for 30")
+        for freq in range(numFreq):
+            uanDatG = readFile(antenna+1, freq+1)
+            thirtyGain = uanDatG[438]
+            uanAngG.append(thirtyGain)
+            #print(thirtyGain)
+        #print("Gain for 45")
+        for freq in range(numFreq):
+            uanDatG = readFile(antenna+1, freq+1) 
+            fortfiveGain = uanDatG[657]
+            uanAngG.append(fortfiveGain)
+            #print(fortfiveGain)
+        #print("Gain for 90")
+        for freq in range(numFreq):
+            uanDatG = readFile(antenna+1, freq+1)
+            nineGain = uanDatG[1314]
+            uanAngG.append(nineGain)
+            #print(nineGain)
```

Comment:
This part seems inefficient, both because of the repeated code and because `readFile` is called multiple times with the same arguments.  It would be better to have a single loop, opening the file once for each `freq`, get all the needed gains for the file, possibly store them in different lists (or in a dict), and then combine them at the end.